### PR TITLE
docs: use issue template forms

### LINKED ISSUE #2271: Turn our issue templates into forms
## Summary 💭

GitHub now makes it easier to require certain fields in issue templates. This would enforce minimal reproductions, so that it reduces maintenance overhead. 

I propose that if a required field value is somehow "cheated" to get it opened, we forcefully close the issue or move it to discussions (we could warn about that in a comment in the template), without further notice. We have to require a minimal effort from our users, as we don't have the capacity to triage half baked reproductions/badly formatted feature requests, etc.

I'm open for suggestions how to do it the most humane way, not discouraging newcomers either.

For further information, see: https://youtu.be/qQE1BUkf2-s

This is up for grab!

### PR DESCRIPTION
**Pull request by mahieyin-rahmun**
## Reasoning 💡
This PR attempts to convert the Github Issue Templates to Issue Forms as per #2271.

## Checklist 🧢

- [x] Documentation
- [ ] Tests
- [ ] Ready to be merged

## Affected issues 🎟

Closes #2271

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yaml
Comment (balazsorban44) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:89]:
```suggestion
  - type: dropdown
    id: pr
    attributes:
      label: Would you be interested to help solve this bug by opening a PR?
      multiple: false
      options:
        - "Yes"
        - "No"
    validations:
      required: true
  - type: markdown
    attributes:
      value: "## Contributing 🙌🏽"
  - type: markdown
    attributes: 
      value: |
        It takes a lot of work 🏋🏻‍♀️ maintaining a library like `next-auth`; any contribution is more than welcome 💚
```

I think the last sentence rather belongs in our `CONTRIBUTING.md`

In addition, my above suggestion would need further tweaking. Is it possible to somehow merge these three sections to one? I think they logically belong together. If not possible, I think the Contributing section should just be a comment, that is not visible in the preview/issue, only when editing.

I would also change the title

> Would you be interested to help solve this bug by opening a PR?

to
 
> ## Contributing 🙌🏽

I am curious about your opinion on this one!

File: .github/ISSUE_TEMPLATE/feature_request.yaml
Comment (balazsorban44) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:24]:
```suggestion
# Thank you very much for reaching out to us regarding the awesome feature that you believe should be included in the NextAuth.js library. Please provide the following information:
  - type: textarea
    id: description
    attributes:
      label: Description 📓
      description: Please provide a more in-depth description of the feature proposed.
    validations:
      required: true
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml
Comment (balazsorban44) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:40]: Should we also turn this into comments as well? That information is going to be a bit noisy for maintainers in my opinion. What do you think?

File: .github/ISSUE_TEMPLATE/feature_request.yaml
Comment (balazsorban44) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:47]:
```suggestion
  - type: textarea
    id: reproduction
    attributes:
      label: How to reproduce ☕️ 
      description: If you have a CodeSandbox playground or some code snippets to help us visualize your idea better, please provide it here.
    validations:
      required: false
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml
Comment (balazsorban44) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:62]: It could be similar to the bug report template, we could ask a yes/no question if they are interested in implementing this feature, and also just refactor it the same way what we land on in the bug report template. I think the last sentence is unnecessary here as well.

File: .github/ISSUE_TEMPLATE/question.yaml
Comment (balazsorban44) [inline .github/ISSUE_TEMPLATE/question.yaml:50]: same as the previous templates above

File: .github/ISSUE_TEMPLATE/question.yaml
Comment (balazsorban44) [inline .github/ISSUE_TEMPLATE/question.yaml:37]:
```suggestion
  - type: textarea
    id: reproduction
    attributes: 
      label: How to reproduce ☕️ 
      description: Please provide a link or code snippets to a minimal reproduction that represents your question
    validations:
      required: false 
```

File: .github/ISSUE_TEMPLATE/typescript.yaml
Comment (balazsorban44) [inline .github/ISSUE_TEMPLATE/typescript.yaml:1]: See the question template

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thank you so much! This looks awesome!

I have some feedback, please report back if you have questions.

I would like us to make the "result" shorter and less noisy, as the current content is although very nice, it might take away a bit from the actual issue by the extra text, which is only useful when filling it out.

I would like to know what you think and feel free to comment on things.

Amazing work so far! :heart:

**Comment by balazsorban44** [comment]
Could you also remove the old `.md` files in this PR? We won't need them, once this is merged :slightly_smiling_face:

File: .github/ISSUE_TEMPLATE/bug_report.yaml
Comment (mahieyin-rahmun) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:30]: Yeah, that would make more sense. For some people, rather than an entire reproduction, a code snippet would suffice.

File: .github/ISSUE_TEMPLATE/bug_report.yaml
Comment (mahieyin-rahmun) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:50]: Makes sense.

File: .github/ISSUE_TEMPLATE/question.yaml
Comment (mahieyin-rahmun) [reply .github/ISSUE_TEMPLATE/question.yaml:37]: Ah, my variable names 🤦

File: .github/ISSUE_TEMPLATE/bug_report.yaml
Comment (mahieyin-rahmun) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:89]: > I think the last sentence rather belongs in our CONTRIBUTING.md

I agree, it felt a bit out of place, but I was doing a raw translation of the .md files to the .yaml files, so I left it there.

> Is it possible to somehow merge these three sections to one? I think they logically belong together. 

I believe Github Forms' yaml syntax doesn't allow mixing markdown sections with regular input sections. I actually tried to have the text from the markdown sections in the description field of input, but it did not like it.

> If not possible, I think the Contributing section should just be a comment, that is not visible in the preview/issue, only when editing.

I believe that is indeed better.

File: .github/ISSUE_TEMPLATE/feature_request.yaml
Comment (mahieyin-rahmun) [reply .github/ISSUE_TEMPLATE/feature_request.yaml:40]: Yes, I also think it serves better as a comment, which would only be visible while filing the issue.

File: .github/ISSUE_TEMPLATE/feature_request.yaml
Comment (mahieyin-rahmun) [reply .github/ISSUE_TEMPLATE/feature_request.yaml:62]: Makes sense.

File: .github/ISSUE_TEMPLATE/bug_report.yaml
Comment (mahieyin-rahmun) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:89]:
```yaml
  - type: markdown
    attributes:
      value: "## Contributing 🙌🏽"
  - type: dropdown
    id: pr
    attributes:
      label: Would you be interested to help solve this bug by opening a PR?
      multiple: false
      options:
        - "Yes"
        - "No"
    validations:
      required: true
  - type: markdown
    attributes: 
      value: |
        It takes a lot of work 🏋🏻‍♀️ maintaining a library like `next-auth`; any contribution is more than welcome 💚
```

This renders:
image

Is this better?

Comment (mahieyin-rahmun) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:89]: I just noticed in the [documentation for Github Issue Forms YAML Syntax](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema#markdown):

> You can use a markdown element to display Markdown in your form that provides extra context to the user, but is not submitted.

File: .github/ISSUE_TEMPLATE/feature_request.yaml
Comment (mahieyin-rahmun) [reply .github/ISSUE_TEMPLATE/feature_request.yaml:40]: I just noticed in the [documentation for Github Issue Forms YAML Syntax](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema#markdown):

> You can use a markdown element to display Markdown in your form that provides extra context to the user, but is not submitted.

**Comment by mahieyin-rahmun** [comment]
@balazsorban44 Thank you for taking the time to review this is such detail! I have made the requested changes. I also replied to a few of your review comments, so kindly check them as well.

I set up a sandbox repository just to see how things render with the current changes, and I attach a few screenshots below. Let me know what you think of them.

`Bug Report Form`
bug_form

`After submitting`
bug_demo

`Feature Request Form`
feature_form

`After submitting`
feature_request_demo

`Question Form`
question_form

`After submitting`
question_demo

`TypeScript Question Form`
typescript_form

`After submitting`
typescript_demo

**Review by 0ubbe — APPROVED** [review]
Wow, amazing work with this ✨  @mahieyin-rahmun @balazsorban44, from my side it seems ready to go 🙌🏽

**Review by balazsorban44 — APPROVED** [review]

**Comment by balazsorban44** [comment]
I love this work! Thank you so much! :pray:

**Comment by balazsorban44** [comment]
@mahieyin-rahmun just a minor thing, but could you please remove the prefixes in titles in a subsequent PR?

I don't think we need them (eg.: [Bug]:), as we have the labels anyway. They take up significant space, especially in the github mobile app, and it's harder to guess the content from the title. 😊

**Pull request merged**
